packages/core/aqua-compiler/src/index.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    generateSources,
+    generateTypes,
+} from './generate/index.js';
+import { CompilationResult } from './generate/interfaces.js';
+
+type OutputType = 'js' | 'ts';
+
+interface JsOutput {
+    sources: string;
+    types: string;
+}
+
+interface TsOutput {
+    sources: string;
+}
+
+type LanguageOutput = JsOutput | TsOutput;
+
+export default async function(res: CompilationResult, outputType: OutputType): Promise<LanguageOutput> {
```

Comment:
I prefer not to use default exports. The good reason for this is that intellisens in IDE doesn't work very good expecially if you have anonymous function like you have here

if you you name function e.g. `aquaToJs` then it would be easy to type this name anywhere and it will be automatically auto-imported, which is very convenient for lib users, I believe